pybamm/models/submodels/electrolyte.py

```diff
@@ -3,180 +3,61 @@
-    #         Flux at the boundaries (Neumann BC).
-    #
-    #     Returns
-    #     -------
-    #     i : array_like, shape (n+1,)
-    #         The current density.
-    #     """
-    #     operators = self.operators[domain]
-    #     kappa_over_c = 1
-    #     kappa = 1
-    #
-    #     # Calculate inner current
-    #     i_inner = kappa_over_c * operators.grad(c) + kappa * operators.grad(e)
-    #
-    #     # Add boundary conditions
-    #     lbc, rbc = current_bcs
-    #     i = np.concatenate([lbc, i_inner, rbc])
-    #
-    #     return i
-    #
-    # def bcs_current(self, domain, t):
-    #     """Boundary conditions for the current conservation equation.
-    #
-    #     Returns
-    #     -------
-    #     2-tuple of array_like, shape(1,)
-    #         The boundary conditions.
-    #
-    #     """
-    #     if domain == "xcn":
-    #         current_bc_left = np.array([0])
-    #         current_bc_right = np.array([self.subparam.icell(t)])
-    #     elif domain == "xcp":
-    #         current_bc_left = np.array([self.subparam.icell(t)])
-    #         current_bc_right = np.array([0])
-    #     return (current_bc_left, current_bc_right)
+import pybamm
+
+
+class StefanMaxwellDiffusion(pybamm.BaseModel):
+    """A class that generates the expression tree for Stefan-Maxwell Diffusion in the
+    electrolyte.
+
+    Attributes
+    ----------
+
+    rhs: dict
+        A dictionary that maps expressions (variables) to expressions that represent
+        the rhs
+    initial_conditions: dict
+        A dictionary that maps expressions (variables) to expressions that represent
+        the initial conditions
+    boundary_conditions: dict
+        A dictionary that maps expressions (variables) to expressions that represent
+        the boundary conditions
+    variables: dict
+        A dictionary that maps strings to expressions that represent
+        the useful variables
+
+    *Extends:* :class:`BaseModel`
+    """
+
+    def __init__(self, G):
+        super().__init__()
+
+        electrolyte_domain = [
+            "negative electrode",
+            "separator",
+            "positive " "electrode",
+        ]
+
+        c_e = pybamm.Variable("c_e", domain=electrolyte_domain)
+
+        delta = pybamm.Parameter("delta")
+        epsilon = pybamm.Parameter("epsilon", domain=electrolyte_domain)
+        b = pybamm.Parameter("b")
+        D_e = pybamm.Parameter("D_e")
+        nu = pybamm.Parameter("nu")
+        t_plus = pybamm.Parameter("t_plus")
+
+        c_e0 = pybamm.Parameter("c_e0")  # Should this be a parameter?
+
+        # Change expression once Binary operations can cope with None input
+        N_e = pybamm.Scalar(0) - D_e * (epsilon * b) * pybamm.Gradient(c_e)
+        # power operator is apparently not defined for class parameters
+
+        self.rhs = {
+            c_e: pybamm.Scalar(0)
+            - pybamm.Divergence(N_e) / delta / epsilon
+            + nu * (pybamm.Scalar(1) - t_plus) * G
```

Comment:
Negate operator implemented in the branch `issue-45-overloaded-binary-ops`; also, `1-t_plus` should work with the fix in that branch (`1` will automatically be converted to `Scalar(1)`)